athena-slack-member-analytics/src/main/java/com/amazonaws/connectors/athena/slack/util/SlackSchemaUtility.java

```diff
@@ -0,0 +1,244 @@
+            }
+        }
+
+        logger.info("getSchemaBuilder: exit");
+        return result;
+    }
+    
+    public static GeneratedRowWriter.RowWriterBuilder getRowWriterBuilder(ReadRecordsRequest recordsRequest, String tableName){
+        logger.info("getRowWriterBuilder: enter");
+        GeneratedRowWriter.RowWriterBuilder builder = GeneratedRowWriter.newBuilder(recordsRequest.getConstraints());
+         
+        logger.info("getRowWriterBuilder: Parsing through json schema"); 
+        JSONObject schema = getSchema(tableName);
+        for (String key : schema.keySet()){
+            // TODO - Handle decimal values,  structs and lists. 
+            // For slack memeber analytics endpoint we only expect integers and strings
+            switch (schema.getJSONObject(key).getString("type")) {
+                case TYPE_INTEGER:
+                    logger.debug("getRowWriterBuilder: key={}, type=Integer", key);
+                    builder.withExtractor(key, (IntExtractor) (Object context, NullableIntHolder value) -> {
+                        value.isSet = 1;
+                        value.value = 0;
+                        if(((JSONObject) context).has(key)) 
+                            value.value = ((JSONObject) context).optInt(key);
+                    });
+                    break;
+                default:
+                    logger.debug("getRowWriterBuilder: key={}, type=String", key);
+                    builder.withExtractor(key, (VarCharExtractor) (Object context, NullableVarCharHolder value) -> {
+                        value.isSet = 1;
+                        value.value = "";
+                        if(((JSONObject) context).has(key))
+                            value.value = ((JSONObject) context).optString(key);
+                    }); 
+            }
+        }
+        logger.info("getRowWriterBuilder: exit");
+        return builder;
+    }
+
+    /**
+     * Use to get the slack token from AWS Secrets manager
+     *
+     * If you need more information about configurations or implementing the
+     * sample code, visit the AWS docs:
+     * https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-samples.html#prerequisites
+     *
+     * @return String Slack authentiation token
+     * @throws Exception if unable to get secret.
+     **/
+    public static String getSlackToken()
+            throws Exception {
+        logger.info("getSlackToken: enter");
+
+        String secretName = System.getenv("secret_name");
+        String region = System.getenv("region");
+
+        if (secretName==null || secretName.isEmpty() || region==null || region.isEmpty())
+            throw new Exception("Missing AWS Secrets environment variables.");
+
+        logger.info("getSlackToken: Retrieving " + secretName);
+
+        // Create a Secrets Manager client
+        AWSSecretsManager client  = AWSSecretsManagerClientBuilder.standard()
+                .withRegion(region)
+                .build();
```

Comment:
MetadataHandler already has built in support for SecretsManager.  see sample below. I'd suggest using that integration instead of this approach here since the provided support includes caching and also client reuse.

https://github.com/awslabs/aws-athena-query-federation/blob/master/athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/MetadataHandler.java#L175

Reply:
implement cachable secrets manager on existing implementation. Existing implementation is returning null for the awssecret when using defaultclient()